transcom/mymove
# B-21168/B-21169 INT Location Lookup component

## [B-21168](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011663)
## [B-21169](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011664)

## Summary

This PR creates a new location lookup component that will be used to search and populate the City State Zip and County fields for the Customer addresses on several pages and the various Office addresses app pages and modals. The fields are being made read only to prevent users from entering bad data into the Address Fields.
An API was added to use to return the results of the search and model was added to use the new v_locations view to query against. 
New addresses added using the city finder will also save/update the us_post_region_cities_id in the addresses table for the address. 

Several pages were modified in the Office App and the Customer Profile creation was updated. The rest of the Customer pages will be updated as part of B-21169 and the Prime Sim will be updated in another backlog.

Is there anything you would like reviewers to give additional scrutiny?

There are many places that were updated to have the new search. Please try and hammer away at any Customer or Office page that uses the <AddressFields> component as there are a LOT of them that were updated. Go through the entire process from making a new Customer to submitting and closing out shipments of all types. Make sure you also include editing of Customer profile and shipment addresses as well as doing this on the Office app side. In short try and break it the best you can. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Make sure you can search by just the Postal Code, or by using City, State Postal Code (Note the , after City that is required to help distinguish between City and State). Make sure you can save and continue once you select a result and fill out the rest of the pages. 

#### NOTE: The City State, Zip and County fields should all be read only when the location lookup component is present. 

Test that the us_post_region_cities_id is populated when you add/edit your addresses. To make this task easier I was using the following query each time I modified an address to make sure the us_post_region_cities_id is updated and is not null or all 0's. Note the like, I started all my street address 1 fields with '123' to make the data easier to view. I also recommend you put the name of the page and/or action you are performing when you modify the address to help you keep track of the changes ids. 

#### select a.street_address_1, a.city, a.state, a.postal_code, a.county, a.us_post_region_cities_id from addresses a where a.street_address_1 like '123%%';

1. Access the Customer app and create a new customer profile
2. Test the Customer Address and the Backup Address pages
3. Create a new shipment of various types and verify you can use the search component and submit the shipments
4. Login to the Office App and make sure you can edit customer profile address, edit shipment addresses using the location lookup tool. Try the various shipment types and edit them or add a new shipment and test the location lookup. Do this as TOO and SC to make sure you can edit addresses or add new shipments. 
5. Try searching for Alaska locations when the Alaska FF is off and when its on. You should only see Alaska results when you enable the Alaska FF.

### Frontend


### Backend


## Screenshots
